### PR TITLE
fix(cli): `cdk watch` constantly prints 'messages suppressed' 

### LINKED ISSUE #18451: (cli): cdk watch shows "...messages supressed..." logs infinitely
### What is the problem?

Hi team, thank you very much for releasing the cool feature to monitor Lambda logs #18159!
I tried it and it's working fantastic, but noticed that it sometimes shows invalid logs infinitely. Details are below.

Could you check it when you get a chance? Thanks. 

### Reproduction Steps

Deploy a function with `cdk watch`, and invoke it.

### What did you expect to happen?

Only valid logs are displayed.

### What actually happened?

After several valid logs, `...messages supressed...` logs continuously appears (say, 1 line per 2-3 seconds per function).

```sh
[/aws/lambda/BackendStack-Handler2AD178334-hzcv8uEytk3K] 1:08:39 PM REPORT RequestId: c680698d-dd1a-4c64-a387-a507203ad9ac      Duration: 493.33 ms      Billed Duration: 494 ms Memory Size: 128 MB     Max Memory Used: 79 MB
[/aws/lambda/BackendStack-Handler2AD178334-hzcv8uEytk3K] 1:08:39 PM (...messages supressed...)
[/aws/lambda/BackendStack-Handler1EB1BC6C8-StOYX6oimaCD] 1:08:38 PM (...messages supressed...)
[/aws/lambda/BackendStack-Handler1EB1BC6C8-StOYX6oimaCD] 1:08:38 PM (...messages supressed...)
[/aws/lambda/BackendStack-Handler1EB1BC6C8-StOYX6oimaCD] 1:08:38 PM (...messages supressed...)
[/aws/lambda/BackendStack-Handler1EB1BC6C8-StOYX6oimaCD] 1:08:38 PM (...messages supressed...)
[/aws/lambda/BackendStack-Handler1EB1BC6C8-StOYX6oimaCD] 1:08:38 PM (...messages supressed...)
[/aws/lambda/BackendStack-Handler2AD178334-hzcv8uEytk3K] 1:08:39 PM (...messages supressed...)
[/aws/lambda/BackendStack-Handler1EB1BC6C8-StOYX6oimaCD] 1:08:38 PM (...messages supressed...)
[/aws/lambda/BackendStack-Handler2AD178334-hzcv8uEytk3K] 1:08:39 PM (...messages supressed...)
[/aws/lambda/BackendStack-Handler1EB1BC6C8-StOYX6oimaCD] 1:08:38 PM (...messages supressed...)
[/aws/lambda/BackendStack-Handler2AD178334-hzcv8uEytk3K] 1:08:39 PM (...messages supressed...)
# ... continues infinitely
```

### CDK CLI Version

2.8.0

### Framework Version

2.8.0

### Node.js Version

16

### OS

macOS

### Language

Typescript

### Language Version

_No response_

### Other information

Considering [this document](https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_FilterLogEvents.html), there is a case that `nextToken` is returned but the response is empty.

> This operation can return empty results while there are more log events available through the token.

Maybe we need to check the length of `response.events` here: (I'm not entirely sure though

https://github.com/aws/aws-cdk/blob/626e6aa1a27feffe7ce60a46a6fdcf26f317eaef/packages/aws-cdk/lib/api/logs/logs-monitor.ts#L200-L209

### PR DESCRIPTION
**Pull request by corymhall**
Currently if `filterLogEvents` returns a nextToken then we print
`(...messages suppressed...)`. According to the docs "This operation
can return empty results while there are more log events available
through the token.".

This fix adds a condition to only print out the suppression message if
`filterLogEvents` returns non-empty results.

Also updates the message to be more descriptive.

fixes #18451

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by rix0rrr — COMMENTED** [review]
lol

**Review by rix0rrr — APPROVED** [review]

**Review by skinny85 — CHANGES_REQUESTED** [review]
Can we actually change the message here?

I think `...messages suppressed...` is too cryptic - customers don't understand what that means.

File: packages/aws-cdk/lib/api/logs/logs-monitor.ts [hunk 1]
```diff
@@ -200,9 +201,9 @@ export class CloudWatchLogEventMonitor {
       // if we have > 100 events let the user know some
       // messages have been supressed. We are essentially
       // showing them a sampling (10000 events printed out is not very useful)
-      if (response.nextToken) {
+      if (filteredEvents.length > 0 && response.nextToken) {
         events.push({
-          message: '(...messages supressed...)',
+          message: '(...messages suppressed...)',
```
Comment (skinny85) [inline packages/aws-cdk/lib/api/logs/logs-monitor.ts:206]: Can we change this to something like:

```suggestion
          message: '>>> `watch` shows only the first 100 log messages - the rest have been truncated...',
```

Feel free to word-smith the exact wording, of course.

Comment (corymhall) [reply packages/aws-cdk/lib/api/logs/logs-monitor.ts]: @skinny85 updated.

**Review by skinny85 — APPROVED** [review]
👍

**Pull request merged**